middleware/sap.go

```diff
@@ -33,6 +33,14 @@ import (
 type SAPService struct {
 	api.Client
 	mutex        sync.Mutex
+
+	authenticateEndpoint *string
```

Comment:
Since proUBC is open-source, and not a commercial product, we can use constants at the top of `SAPService` to specify default routes...